ONSdigital/eq-schema-validator
# Define new variant schema

### What is the context of this PR?
Card: https://trello.com/c/3r1PVjpm/2753-update-json-schema-and-validator-for-new-variant-schema
This change implements the rules and restrictions of the new variant schema.
- Allows questions to be defined as `questions` or `variants`.

### How to review

- Ensure changes are as per [spec](https://github.com/ONSdigital/eq-schema-validator/blob/bb7e852fb81ae73cfc7d311284a2ec3b14df741d/doc/decisions/0002-define-proxy-mechanism.md).

- Ensure variants can be used by types like `Interstitials`, `Confirmation` as well as `Question` types.

Was unsure if the naming should be changed to `question_variants` and `content_variants`.
